### PR TITLE
[irods/irods#6953] irepl: Deprecate "backupMode" (main)

### PR DESCRIPTION
**Pull request by alanking**
Addresses irods/irods#6953

Companion PR: https://github.com/irods/irods/pull/7591

### CODE REVIEW

File: src/irepl.cpp [hunk 1]
```diff
@@ -160,7 +164,7 @@ usage() {
         " ",
         "Options are:",
         " -a  all - if used with -U, update all stale copies",
-        " -B  Backup mode - if a good copy already exists in this",
+        " -B  Backup mode - (Deprecated) if a good copy already exists in this",
```
Comment (trel) [inline src/irepl.cpp:167]: hesitate to suggest this - but consider changing the word 'copy' on this line and the next to be 'replica'... but ... not even sure that is correct/true.

ignore with impunity.

Comment (alanking) [reply src/irepl.cpp:167]: I think... I'm going to leave this as-is. Copy might actually be the right word in this case - I simply do not know.

Comment (trel) [reply src/irepl.cpp:167]: Defer.  Resolving.

File: src/irepl.cpp [hunk 2]
```diff
@@ -96,6 +96,10 @@ main( int argc, char **argv ) {
         exit( 7 );
     }
 
+    if (0 != myRodsArgs.backupMode) {
+        printf("Warning: Backup Mode (-B) is deprecated. Its use should be avoided.\n");
```
Comment (MartinFlores751) [inline src/irepl.cpp:100]: Can `std::printf(...)`.

**Review by MartinFlores751 — COMMENTED** [review]
Bonus item.

File: src/irepl.cpp (hunk 2 above)
Comment (alanking) [reply src/irepl.cpp]: Done

File: src/irepl.cpp [hunk 3]
```diff
@@ -96,6 +96,10 @@ main( int argc, char **argv ) {
         exit( 7 );
     }
 
+    if (0 != myRodsArgs.backupMode) {
+        std::printf("Warning: Backup Mode (-B) is deprecated. Its use should be avoided.\n");
+    }
```
Comment (korydraughn) [inline src/irepl.cpp:101]: Do we print out messages about flags being deprecated for other icommands and options?

Comment (alanking) [reply src/irepl.cpp:101]: Yes.

https://github.com/irods/irods/issues/4860
https://github.com/irods/irods/issues/7498

**Comment by alanking** [comment]
Removed warning message

**Review by korydraughn — APPROVED** [review]
Squash/Pound when ready.

**Comment by alanking** [comment]
Squashed. Will # along with https://github.com/irods/irods/pull/7591

**Review by korydraughn — APPROVED** [review]
Pound when ready.

**Comment by alanking** [comment]
Done. Mergin...

**Pull request merged**